### PR TITLE
Return "parameterId" as "id" in removeParameter action

### LINKED ISSUE #33826: JavaScript TypeError when removing dashboard filter
### Describe the bug

https://github.com/metabase/metabase/assets/6830683/0050d589-dd93-4387-a086-8d2f957082ab

### To Reproduce

1. Create a new dashboard
2. Add a filter to it
3. Remove the filter
4. There's an error in JS console

### Expected behavior

No JS error in the console

### Logs

[screenshot of code location](https://github.com/metabase/metabase/assets/6830683/871ca02d-ec52-4b2d-a94e-4bff75b034d8)

```
app-main.hot.bundle.js?9216a7efeb1855327899:510058 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'id')
    at next (app-main.hot.bundle.js?9216a7efeb1855327899:510058:29)
    at vendor.hot.bundle.js?44b07ed40a65ac2f757f:299380:64
    at vendor.hot.bundle.js?44b07ed40a65ac2f757f:299794:22
    at Array.reduce (<anonymous>)
    at __WEBPACK_DEFAULT_EXPORT__ (vendor.hot.bundle.js?44b07ed40a65ac2f757f:299793:21)
    at vendor.hot.bundle.js?44b07ed40a65ac2f757f:299430:12
    at combination (vendor.hot.bundle.js?44b07ed40a65ac2f757f:34352:29)
    at vendor.hot.bundle.js?44b07ed40a65ac2f757f:299236:22
    at Array.reduce (<anonymous>)
    at __WEBPACK_DEFAULT_EXPORT__ (vendor.hot.bundle.js?44b07ed40a65ac2f757f:299231:21)
```

### Information about your Metabase installation

v47.1, `master`, ca7d434

### Severity

P2

### PR DESCRIPTION
**Pull request by ibedwi**
Closes https://github.com/metabase/metabase/issues/33826

#### Root cause

The `removeParameter` action which used in `REMOVE_PARAMETER` action handler in the reducers never return any `payload`. But the `REMOVE_PARAMETER` action handler in the `reducers` expect to retrieve an object with an `id` property.

#### How to solve

Return the `parameterId` as `id` in the `removeParameter` action.

### How to verify

1. Create a new dashboard
2. Add a filter to it
3. Remove the filter
4. There's an error in JS console

### Demo

https://github.com/metabase/metabase/assets/15979404/90fabea8-aff4-4e06-9686-2d16cb77f7b1

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Review by kamilmielnik — COMMENTED** [review]
Fix looks good!
I left some comments about tests.

Thank you for another contribution :heart:

File: frontend/src/metabase/dashboard/reducers.unit.spec.js [hunk 1]
```diff
@@ -188,6 +201,25 @@ describe("dashboard reducers", () => {
...
+      expect(
+        reducer(
+          {
+            ...initState,
+            parameterValues: {
+              123: null,
+              456: null,
+            },
+          },
+          await removeParameter(123)(dispatch, getState),
```
[the remark this answers is not in the document]
Comment (ibedwi) [reply frontend/src/metabase/dashboard/reducers.unit.spec.js]: Thanks for your explanation!

File: frontend/src/metabase/dashboard/actions/parameters.unit.spec.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        },
+      },
+    });
+  });
+  it("should return the `parameterId` as `payload.id` (metabase#34680)", async () => {
+    const result = await removeParameter(123)(dispatch, getState);
+    expect(result).toEqual({
+      type: REMOVE_PARAMETER,
+      payload: { id: 123 },
+    });
```
Comment (ibedwi) [inline frontend/src/metabase/dashboard/actions/parameters.unit.spec.js:29]: @kamilmielnik I've updated the unit test as you suggested here 👍

**Review by kamilmielnik — DISMISSED** [review]

File: frontend/src/metabase/dashboard/actions/parameters.unit.spec.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+          },
+        },
+        parameterValues: {
+          123: null,
+          456: null,
+        },
+      },
+    });
+  });
+  it("should return the `parameterId` as `payload.id` (metabase#34680)", async () => {
```
Comment (kamilmielnik) [inline frontend/src/metabase/dashboard/actions/parameters.unit.spec.js:24]: 1. It should be issue id, not PR id. Apologies for my [misleading suggestion](https://github.com/metabase/metabase/pull/34680#discussion_r1360214312).
2. Nit: we always put new lines here

```suggestion

  it("should return the `parameterId` as `payload.id` (metabase#33826)", async () => {
```

File: frontend/src/metabase/dashboard/reducers.unit.spec.js [hunk 4]
```diff
@@ -188,6 +201,25 @@ describe("dashboard reducers", () => {
         ),
       ).toEqual({ ...initState, parameterValues: { 456: "def" } });
     });
+    it("should remove the parameter from `parameterValues`", async () => {
```
[the remark this answers is not in the document]
Comment (kamilmielnik) [reply frontend/src/metabase/dashboard/reducers.unit.spec.js]: Oh, sorry, I meant the issue id, i.e. #33826

**Review by kamilmielnik — COMMENTED** [review]
Almost ready to be merged :pray:

File: frontend/src/metabase/dashboard/actions/parameters.unit.spec.js (hunk 3 above)
Comment (ibedwi) [reply frontend/src/metabase/dashboard/actions/parameters.unit.spec.js]: updated! thanks for the feedback. I'll take a note for the issue id things. Thanks for your guidance!

**Review by kamilmielnik — APPROVED** [review]
Looks good!
Thank you for the contribution :rocket:

File: frontend/src/metabase/dashboard/actions/parameters.unit.spec.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import { removeParameter, REMOVE_PARAMETER } from "./parameters";
+
+describe("removeParameter", () => {
+  let dispatch;
+  let getState;
+  beforeEach(() => {
+    dispatch = jest.fn();
+    getState = () => ({
```
Comment (uladzimirdev) [inline frontend/src/metabase/dashboard/actions/parameters.unit.spec.js:8]: shouldn't we prepare real type safe state? @kamilmielnik

Comment (kamilmielnik) [reply frontend/src/metabase/dashboard/actions/parameters.unit.spec.js:8]: Ideally, yes.
But I don't think we have a single unit test with real type safe state yet, do we?

BTW this test uses state definition almost identical to another one from existing test for action creators: https://github.com/metabase/metabase/blob/8fb7a7842528197567c523ae921b79ce055d3bf9/frontend/src/metabase/dashboard/actions/actions.unit.spec.js#L26-L36

**Pull request merged**